ocaml/parsing/pprintast.ml

```diff
@@ -864,15 +862,30 @@ and expression ?(jane_syntax_parens = false) ctxt f x =
       | Pexp_letexception _ | Pexp_letop _
         when ctxt.semi ->
         paren true (expression reset_ctxt) f x
-    | Pexp_fun (l, e0, p, e) ->
-        pp f "@[<2>fun@;%a@;%a@]"
-          (label_exp ctxt) (l, e0, p)
-          (pp_print_pexp_function ctxt "->") e
     | Pexp_newtype (lid, e) ->
         pp f "@[<2>fun@;(type@;%s)@;%a@]" lid.txt
           (pp_print_pexp_function ctxt "->") e
-    | Pexp_function l ->
-        pp f "@[<hv>function%a@]" (case_list ctxt) l
+    | Pexp_function (params, constraint_, body) ->
+        begin match params, constraint_ with
+          (* Omit [fun] if there are no params. *)
+          | [], None ->
+              let should_paren =
```

Comment:
Looks like upstream has another comment here that you've omitted

Reply:
Yep, thanks.